Api/V8/OAuth2/Repository/AuthCodeRepository.php

```diff
@@ -0,0 +1,101 @@
+<?php
+namespace Api\V8\OAuth2\Repository;
+
+use Api\V8\BeanDecorator\BeanManager;
+use Api\V8\OAuth2\Entity\AuthCodeEntity;
+use League\OAuth2\Server\Entities\AuthCodeEntityInterface;
+use League\OAuth2\Server\Repositories\AuthCodeRepositoryInterface;
+
+class AuthCodeRepository implements AuthCodeRepositoryInterface
+{
+    /**
+     * @var BeanManager
+     */
+    private $beanManager;
+
+    /**
+     * @param AuthCodeEntity $authCodeEntity
+     * @param BeanManager $beanManager
+     */
+    public function __construct(BeanManager $beanManager)
+    {
+        $this->beanManager = $beanManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function persistNewAuthCode(AuthCodeEntityInterface $authCodeEntity)
+    {
+        // Some logic to persist the auth code to a database
+
+        $clientId = $authCodeEntity->getClient()->getIdentifier();
+
+        /** @var \OAuth2Clients $client */
+        $client = $this->beanManager->getBeanSafe('OAuth2Clients', $clientId);
```

Comment:
The ```$client``` variable doesn't appear to be used anywhere.

Reply:
Removed.